controllers/node_controller.go

```diff
@@ -156,24 +142,9 @@ func (r *NodeReconciler) doFinalize(ctx context.Context, log logr.Logger, node *
 }
 
 func (r *NodeReconciler) cleanupLogicalVolume(ctx context.Context, log logr.Logger, lv *topolvmv1.LogicalVolume) error {
-	finExists := false
-	for _, fin := range lv.Finalizers {
-		if fin == topolvm.GetLogicalVolumeFinalizer() {
-			finExists = true
-			break
-		}
-	}
-
-	if finExists {
+	if controllerutil.ContainsFinalizer(lv, topolvm.GetLogicalVolumeFinalizer()) {
```

Comment:
`RemoveFinalizer` tells us whether the finalizer existed or not.
https://github.com/kubernetes-sigs/controller-runtime/blob/c304e7ec2ee7879f53ce69c12a7aa133853fccdb/pkg/controller/controllerutil/controllerutil.go#L366